code/game/objects/structures/roof.dm

```diff
@@ -34,7 +34,7 @@
 			neighbor = locate() in adjacent_loc
 			if(!neighbor)
 				neighbor = new(adjacent_loc)
-	return INITIALIZE_HINT_LATELOAD
+	return INITIALIZE_HINT_ROUNDSTART
```

Comment:
I don't imagine lazy_nodes will work correctly if a nightmare insert modifies a roof. You really ought to figure out how to get them to load after roundstart.

If the issue is that master lateloads before the nodes lateload, have the nodes when they're done loading call a proc to the master node that increments a ready count in the master node.

If the master gets signaled a node is ready, and the count of ready nodes (calls to this proc) now matches the count of linked nodes, then master initializes. You just need to ensure that the count of linked nodes will never be incorrect.